### ISSUE TITLE
TextNote does not call applyStyle() before draw()

### ISSUE THREAD

**Issue by mscuthbert** [0xfe/vexflow#602]
Should this be happening?  I can't seem to apply fillStyle or strokeStyle to text, which leaves my text in the same color as the last note drawn.  I can make a fix, but just making sure this isn't omitted for a reason.

**Comment by 0xfe** [comment]
Definitely not intentional.

**Comment by bneumann** [comment]
I have the problem, that when i change the style for 1 note it changes it for all the following PLUS some other elements as stave bars and braces. 

Here it looks almost ok (except for the bar, as i only apply the style to the notes):
![image](https://user-images.githubusercontent.com/2605131/36934075-ccfc6076-1ee3-11e8-88d1-1b9f7bb8cc10.png)

And my next line is completely wrongly colored:
![image](https://user-images.githubusercontent.com/2605131/36934087-f8460cbe-1ee3-11e8-834b-3a469d305aca.png)

@mscuthbert I linked your changes into my project (working for [OSMD](https://github.com/opensheetmusicdisplay/opensheetmusicdisplay) right now) but it is the same effect. However if I set the style manually into the context before calling the draw method it works just fine.

So in a nutshell:
- vexflow's setStyle seems to be affecting other elements too
- Setting the style in the context before the draw and restoring it later manually works

**Comment by mscuthbert** [comment]
I've seen the same behavior -- I didn't know if it was intentional (I'm not so up on the Context classes) -- I've generally avoided it by setting a style object on everything I put out, usually set to black.

**Comment by gristow** [comment]
I'm working on this now... it's almost certainly something my code introduced, should have a PR shortly.

**Comment by gristow** [comment]
Ok -- this is fixed in #604.

**Comment by mscuthbert** [comment]
Thanks @gristow  -- I think that the issue with TextNote style not being applies is separate from the other PR -- would you consider reviewing this PR also?  Thanks!

**Comment by bneumann** [comment]
Thanks, awesome. Do you have an ETA for the next release on npm? Just curious

**Comment by bneumann** [comment]
Hi @gristow I checked out the newest master and tried it and it looks better but it seems to be randomly coloring my notes. Since I already set up everything I will have a look into it.

**Comment by gristow** [comment]
@bneumann please let me know what you find -- I walked through all of the styling & drawing code quite closely for `StaveNote` `StemmableNote` `Stem` `Element` and `Note` pretty carefully yesterday, and colors seem to be setting & restoring properly in existing tests and new ones I'll push shortly. But if you can share code in which it's not functioning, I'd love to build a test for that.

**Comment by bneumann** [comment]
So what I found is, that if I comment the applyStyle and restoreStyle in the stavenote.js it works almost as expected:

```javascript    // Apply the overall style -- may be contradicted by local settings:
    // this.applyStyle();
    this.setAttribute('el', this.context.openGroup('stavenote', this.getAttribute('id')));
    this.context.openGroup('note', null, { pointerBBox: true });
    if (shouldRenderStem) this.drawStem();
    this.drawNoteHeads();
    this.drawFlag();
    this.context.closeGroup();
    this.drawModifiers();
    this.context.closeGroup();
    // this.restoreStyle();
    this.setRendered();
```
![image](https://user-images.githubusercontent.com/2605131/36948118-eff98b9c-1fd5-11e8-9e49-0393a1ff114c.png)

Without the change it looks like this:

![image](https://user-images.githubusercontent.com/2605131/36948130-1e909eaa-1fd6-11e8-9649-861eddb70e75.png)

I tried removing parts of my code as well and figured that calling ```voice.draw(ctx, stave)``` seems to cause this behaviour.

Coloring single notes, beams, whatever does not seem to be the problem. But the voice somehow misses to reset the context?!

**Comment by gristow** [comment]
I can't seem to replicate this when using multiple voices and staves -- [here's a fiddle](https://jsfiddle.net/gristow/d1vf090n/5/) using the latest build to start with. Can you fork the fiddle & replicate the problem in it?
